### ISSUE TITLE
Liskov violation in method argument names not detected

### ISSUE THREAD

**Issue by jonathanslenders** [python/mypy#17606]
**Feature**

Detect when argument names from a subclass are incompatible with the parent class.

**Pitch**

For keyword-only arguments, mypy does detect when argument names of a subclass differ from the parent.
For positional-only, names don't matter obviously, but I believe when parameters can be call either as a positional or keyword argument, the Liskov principle applies.

```python
﻿﻿class A:
    def f(self, a: int) -> None: ...

class B(A):
    def f(self, b: int) -> None: ...
```

I couldn't find any open issue, or a setting for enabling detecting this kind of bug.

**Comment by jairov4** [comment]
Sounds reasonable to me, often we use pass named parameters for positionals in order to make more clear the reason for some literals.

**Comment by tmke8** [comment]
Duplicate of #6709, I believe.

Originally, this was intentional behavior because there was no way to specify positional-only arguments before Python 3.8, but Python 3.7 is EoL, so this really should be changed now.
